soc/arm/nuvoton_npcx/npcx4/soc.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <soc.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
+
+static int soc_init(void)
+{
+
+	return 0;
+}
+
+SYS_INIT(soc_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
is this needed? the other ones have a soc_init as well and it's empty, may as well drop them all.

Reply:
After looking at the other soc drivers, it might be a good idea to move some workaround methods only for the soc series to here. For example, moving [line 154](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nuvoton_npcx/common/scfg.c#L151) of scfg.c to [npcx7/soc.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nuvoton_npcx/npcx7/soc.c#L15). And if there's no workaround for this series, drop soc.c file directly.

If you're okay, I will submit the other PR for code review.